bus-mapping/src/eth_types.rs

```diff
@@ -0,0 +1,485 @@
+//! Ethereum types used to deserialize responses from web3 / geth.
+
+use crate::evm::{memory::Memory, stack::Stack, storage::Storage};
+use crate::evm::{Gas, GasCost, OpcodeId, ProgramCounter};
+use pasta_curves::arithmetic::FieldExt;
+use serde::{de, Deserialize, Serialize};
+use std::collections::HashMap;
+use std::str::FromStr;
+pub use web3::types::{self, AccessList, Bytes, Index, H2048, H64, U256, U64};
+
+use subtle::CtOption;
+
+/// Trait used to define types that can be converted to a 256 bit scalar value.
+pub trait ToScalar<F: FieldExt> {
+    /// Convert the type to a scalar value.
+    fn to_scalar(&self) -> CtOption<F>;
+}
+
+/// Trait used to convert a type to a [`Word`].
+pub trait ToWord {
+    /// Convert the type to a [`Word`].
+    fn to_word(&self) -> Word;
+}
+
+/// Trait uset do convert a scalar value to a 32 byte array in big endian.
+pub trait ToBigEndian {
+    /// Convert the value to a 32 byte array in big endian.
+    fn to_be_bytes(&self) -> [u8; 32];
+}
+
+// We use our own declaration of another U256 in order to implement a custom deserializer that can
+// parse U256 when returned by structLogs fields in geth debug_trace* methods, which don't contain
+// the `0x` prefix.
+#[allow(clippy::all)]
+mod uint_types {
+    uint::construct_uint! {
+        /// 256-bit unsigned integer.
+        pub struct DebugU256(4);
+    }
+}
+pub use uint_types::DebugU256;
+
+impl<'de> Deserialize<'de> for DebugU256 {
+    fn deserialize<D>(deserializer: D) -> Result<DebugU256, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        DebugU256::from_str(&s).map_err(de::Error::custom)
+    }
+}
+
+impl Serialize for DebugU256 {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let s = format!("{:#x}", self);
+        if s == "0x00" {
+            serializer.serialize_str("0x0")
+        } else {
+            serializer.serialize_str(&s)
+        }
+    }
+}
+
+impl<F: FieldExt> ToScalar<F> for DebugU256 {
+    fn to_scalar(&self) -> CtOption<F> {
```

Comment:
As we discussed `CtOption` API is not the best one to have. Specially if we're not developing anything that needs to be ctant time.

I would talk to @miha-stopar and @han0110 and consider returning and using `Option`.

`CtOption` contains an Into conversion to it, so should be trivial to adapt it, and would give us much more flexibility.
I made that PR IIRC: https://github.com/dalek-cryptography/subtle/pull/75

Reply:
I've updated the PR to use `Option` rather than `CtOption` following your advice (using the `into()` that you introduced in subtle :P).  Currently this is only used in zkevm-circuits for tests and always with `unwrap()`, so both `Option` and `CtOption` compile fine.